Mathlib/CategoryTheory/CodiscreteCategory.lean

```diff
@@ -0,0 +1,139 @@
+/-
+Copyright (c) 2024 Alvaro Belmonte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alvaro Belmonte
+-/
+import Mathlib.CategoryTheory.EqToHom
+import Mathlib.CategoryTheory.Pi.Basic
+import Mathlib.Data.ULift
+import Mathlib.CategoryTheory.Category.Cat
+import Mathlib.CategoryTheory.Adjunction.Basic
+
+/-!
+# Codiscrete categories
+
+We define `Codiscrete A` as an alias for the type `A` ,
+and use this type alias to provide a `Category` instance
+whose Hom type are Unit types.
+
+`Codiscrete.lift` promotes a function `f : C → A` (for any category `C`) to a functor
+`Discrete.functor f : C ⥤  Codiscrete A`.
+
+Similarly, `Codiscrete.natTrans` and `Codiscrete.natIso` promote `I`-indexed families of morphisms,
+or `I`-indexed families of isomorphisms to natural transformations or natural isomorphism.
+
+We define `functor : Type u ⥤ Cat.{0,u}` which sends a type to the codiscrete category and show
+it is right adjoint to `Cat.objects.`
+-/
+namespace CategoryTheory
+
+universe u v w
+/-- We start by introducting an alias for the type underlying a codiscrete category
+structure.-/
```

Comment:
The comment `/-- ... -/` just before a definition is going to appear in the user interface (typically in VSCode) when we hover the pointer over `Codiscrete`: then, as much as possible, we want this to be a description of the definition.

```suggestion

/-- The type of objects `Codiscrete A` of a codiscrete category is a type synonym for `A`. -/
```